src/software/geom/util.cpp

```diff
@@ -427,27 +347,23 @@ std::optional<Segment> mergeFullyOverlappingSegments(Segment segment1, Segment s
 {
     // If the segments are not parallel, then return std::nullopt. (The segments are
     // parallel if all points are collinear)
-    if (!collinear(segment1, segment2))
+    if (!segment1.collinear(segment2))
     {
         return std::nullopt;
     }
 
-    Segment largest_segment, smallest_segment;
-    // Grab the largest segment
-    if (segment1.toVector().lengthSquared() > segment2.toVector().lengthSquared())
-    {
-        largest_segment  = segment1;
-        smallest_segment = segment2;
-    }
-    else
+    Segment largest_segment  = segment1;
+    Segment smallest_segment = segment2;
+    // Swap if initialized wrong
```

Comment:
über mega nit: just checked the spec, its technically not wrong to provide the largest segment first, because the spec only says "first segment" and "second segment"

Reply:
I'm confused by this...

Reply:
the comment says "Swap if initialized wrong", its not to wrong to initialize the other way

again this is an uber mega nit, so don't bother